awx/ui_next/src/screens/Organization/shared/OrganizationForm.test.jsx

```diff
@@ -70,91 +70,117 @@ describe('<OrganizationForm />', () => {
       );
     });
 
-    expect(OrganizationsAPI.readInstanceGroups).toHaveBeenCalled();
-    expect(wrapper.find('OrganizationForm').state().instanceGroups).toEqual(
-      mockInstanceGroups
+    await waitForElement(
+      wrapper,
+      'InstanceGroupsLookup',
+      el => el.length === 1
     );
+    expect(OrganizationsAPI.readInstanceGroups).toHaveBeenCalled();
+    expect(wrapper.find('InstanceGroupsLookup Chip span')).toHaveLength(2);
   });
 
-  test('changing instance group successfully sets instanceGroups state', async () => {
+  test('Instance group is rendered when added', async () => {
+    OrganizationsAPI.readInstanceGroups.mockReturnValue({
+      data: { results: [] },
+    });
     let wrapper;
     await act(async () => {
       wrapper = mountWithContexts(
         <OrganizationForm
           organization={mockData}
-          handleSubmit={jest.fn()}
-          handleCancel={jest.fn()}
+          onSubmit={jest.fn()}
+          onCancel={jest.fn()}
           me={meConfig.me}
         />
       );
     });
-
-    const lookup = wrapper.find('InstanceGroupsLookup');
+    const lookup = await waitForElement(
+      wrapper,
+      'InstanceGroupsLookup',
+      el => el.length === 1
+    );
     expect(lookup.length).toBe(1);
-
-    lookup.prop('onChange')(
-      [
-        {
-          id: 1,
-          name: 'foo',
-        },
-      ],
-      'instanceGroups'
+    expect(lookup.find('Chip span')).toHaveLength(0);
+    await act(async () => {
+      lookup.prop('onChange')(
+        [
+          {
+            id: 1,
+            name: 'foo',
+          },
+        ],
+        'instanceGroups'
+      );
+    });
+    const group = await waitForElement(
+      wrapper,
+      'InstanceGroupsLookup Chip span',
+      el => el.length === 1
     );
-    expect(wrapper.find('OrganizationForm').state().instanceGroups).toEqual([
-      {
-        id: 1,
-        name: 'foo',
-      },
-    ]);
+    expect(group.text()).toEqual('foo');
   });
 
-  test('changing inputs should update form values', async () => {
+  test('changing inputs and saving triggers expected callback', async () => {
     let wrapper;
+    const onSubmit = jest.fn();
     await act(async () => {
       wrapper = mountWithContexts(
         <OrganizationForm
           organization={mockData}
-          handleSubmit={jest.fn()}
-          handleCancel={jest.fn()}
+          onSubmit={onSubmit}
+          onCancel={jest.fn()}
           me={meConfig.me}
         />
       );
     });
-
-    const form = wrapper.find('Formik');
-    wrapper.find('input#org-name').simulate('change', {
-      target: { value: 'new foo', name: 'name' },
+    await waitForElement(wrapper, 'ContentLoading', el => el.length === 0);
+    await act(async () => {
```

Comment:
I combined a few of the redundant `onSubmit...` tests along with the input-updating test into one. Coverage is the same.